database/engine/metadata_log/compaction.c

```diff
@@ -123,9 +129,11 @@ static void compact_record_by_uuid(struct metalog_instance *ctx, uuid_t *uuid)
             }
             break;
         case GUID_TYPE_HOST:
-            if (ctx->current_compaction_id > host->compaction_id) {
-                host->compaction_id = ctx->current_compaction_id;
-                buffer = metalog_update_host_buffer(host);
+            error_with_guid(uuid, "Compaction: Processing HOST record");
+            RRDHOST *this_host = metalog_get_host_from_uuid(ctx, uuid);
```

Comment:
You don't know here if `this_host` is `NULL` or even a different pointer than `host`. How do we resolve this? Maybe fallback to the old code for now, so that we can merge to master?

Reply:
Eventually it could be a different host (null would be an error).  I can skip the call and use `ctx->rrdeng_ctx->host` to fallback to the old code.